src/common/mongo_dao.py

```diff
@@ -756,6 +760,43 @@ def search_node(self, data_commons, node_type, node_id):
             self.log.exception(e)
             self.log.exception(f"Failed to find release record for {data_commons}/{node_type}/{node_id}: {get_exception_msg()}")
             return False
+        
+    def search_node_by_study(self, studyID, entity_type, node_id):
+        """
+        Search release collection for given node, if not found, search it in dataRecord collection
+        :param studyID:
+        :param node_type:
+        :param node_id:
+        :return:
+        """
+        db = self.client[self.db_name]
+        data_collection = db[RELEASE_COLLECTION]
+        try:
+            submissions = self.find_submissions({STUDY_ID: studyID})
+            if len(submissions) < 2:  #if there is only one submission that's own submission, skip.
+                return None
+            submission_id_list = [item[ID] for item in submissions]
+            results = data_collection.find({NODE_TYPE: entity_type, NODE_ID: node_id, SUBMISSION_ID: {"$in": submission_id_list}})
```

Comment:
I don't think searching entity type in node type will work. Since entity type is an abstract concept and probably won't be same as any node types. 

I'd suggest adding a **entityType** to **dataRecord** when loading data into DB, then search **entityType** in this function.

Reply:
the nodeType here is entity type resolved in the data model dict.

Reply:
If need to add entityType into dataRecord, it need to be added to release also.  That need lots of code changes.   Please create a separate task.  I'll put the ticket in hold.

Reply:
Updated the search function.